lib/ansible/constants.py

```diff
@@ -183,3 +182,7 @@ def set_constant(name, value, export=vars()):
         value = ensure_type(value, setting.name)
 
     set_constant(setting.name, value)
+
+    # speical hack to ensure all local temp files use the configured temp dir by default
+    if setting.name == 'DEFAULT_LOCAL_TMP':
+        os.environ['TEMPDIR'] = os.environ['TEMP'] = os.environ['TMP'] = value
```

Comment:
This is not a good idea as it could break things that need a system tempdir instead of a per-user tempdir.

Reply:
on controller side? i looked at all our tempfile uses and didn't see a case that would fit that scenario.
on remote we do have code that handles that case.

Reply:
Yeah, on controller-side.  for instance, if we have the local connection plugin with become to an unprivileged user, this might break any module that uses tempfile. (unless the become command always sets the environment variables to the value the suer we're becoming would expect.  But for sudo, at least, I think we let the user determine that via ansible.cfg settings).

Reply:
There's other places that could break too although those are more esoteric.  (We allow users to write their own plugins and changing TEMPDIR can affect the expectations of the tools that they might be invoking from the plugins).  It's the interaction of various different become commands with localhost that has me worried..

Reply:
Agreed today that we'll remove this and fix the callers to use 
`dir=C.DEFAULT_LOCAL_TMP`

Should be here:
```
 grep -ri tempfile . |grep import|grep -v ./modules|grep -v ./module_utils
./playbook/role/requirement.py:import tempfile
./executor/module_common.py:import tempfile
./executor/task_queue_manager.py:import tempfile
./plugins/action/__init__.py:import tempfile
./plugins/action/assemble.py:import tempfile
./plugins/action/copy.py:import tempfile
./plugins/action/template.py:import tempfile
./plugins/action/win_copy.py:import tempfile
./plugins/connection/winrm.py:import tempfile
./plugins/connection/paramiko_ssh.py:import tempfile
./plugins/connection/funcd.py:import tempfile
./galaxy/role.py:import tempfile
./parsing/dataloader.py:import tempfile
./parsing/vault/__init__.py:import tempfile
./config/manager.py:import tempfile
```